### PR TITLE
Replace Travis by GitHub actions & add newer versions to build matrix

### PR DESCRIPTION
**Pull request by browniebroke**
- Replace Travis by GitHub actions
- Update badges in README and documentation
- Add Python 3.8+3.9+3.10, Django 3.1+3.2 and DRF 3.12 to the build matrix

On thing to note, the test result don't report as pull request status, this is a limitation for security reasons of GitHub actions: a new workflow added from a fork will not run on `pull_request`. You can see [the build status on my fork](https://github.com/browniebroke/django-rest-framework-serializer-extensions/actions/workflows/test.yml).

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,32 +1,45 @@
...
-       pytest-cov==2.6.1
-       hashids==1.1.0
-       mock==2.0.0
+       drf312: djangorestframework~=3.12.0
+       django-test-plus==2.2.0
+       pytest==6.2.5
+       pytest-django==4.4.0
+       pytest-cov==3.0.0
+       hashids==1.3.1
+       mock==4.0.3
```
Comment (browniebroke) [inline tox.ini:37]: I had to update `pytest` for Python 3.10 support. I took this opportunity to update all pinned versions here.

File: tox.ini [hunk 2]
```diff
@@ -1,32 +1,45 @@
...
 
 [testenv:py38-flake8]
 commands = ./runtests.py --lintonly
 deps =
-       pytest==3.0.5
+       pytest==6.2.5
        flake8==3.7.7
 
 [testenv:py38-codecov]
```
Comment (browniebroke) [inline tox.ini:45]: I think this is not running right now. In my experience, it's a bit tricky to test coverage is properly reported from a fork... Probably bets to use [the dedicated action](https://github.com/codecov/codecov-action) for this.

File: tox.ini [hunk 3]
```diff
@@ -1,44 +1,47 @@
...
 
-[testenv:py38-codecov]
-commands =
-       {[testenv]commands}  # Run the tests and generate coverage report
-       codecov -e TOX_ENV   # Post coverage stats to codecov.io
-deps =
-       {[testenv]deps}
-       Django==3.0.0
-       djangorestframework==3.10.3
-       codecov==2.0.15
```
Comment (browniebroke) [inline tox.ini:40]: Replaced this by the official action to report coverage: https://github.com/codecov/codecov-action

**Comment by evenicoulddoit** [comment]
Hey, thanks so much for this. Overall looks great. Not used GH Actions before, so I'm going to look into this, but it seems like prior to merging this I'll also have to resolve the coverage reporting. I'd also like to maintain a "current - 2" stance on support for Python, Django etc., so that would mean dropping some of the older versions _in addition_ to adding support for new ones.

**Comment by browniebroke** [comment]
I _think_ [the coverage reporting is now working](https://github.com/browniebroke/django-rest-framework-serializer-extensions/runs/4068969440?check_suite_focus=true#step:6:31).

**Comment by browniebroke** [comment]
> I'd also like to maintain a "current - 2" stance on support for Python, Django etc.

Hum, yes that makes sense for Django & DRF. Does supporting Python 3.6-3.10 requires a lot of effort? I think 3.6 goes EOL in a [little less than 2 months](https://endoflife.date/python) so may I suggest to maybe put out a last release with support added for newer versions before dropping it when it goes out of support?

**Comment by evenicoulddoit** [comment]
Hey @browniebroke thanks again for doing this. I finally got some time to come back to this project. I cleaned it up somewhat, and at the same time I integrated GH Actions, basically copy-pasta your work from here. It was unfortunate that it only seems triggered on my own branches, but :shrug: 

I'm going to try to look into updating the supported versions. Between you doing this work and me picking this up today though, it seems like Django 4.0 was released, which has some breaking changes, so I'll need to address those too

**Comment by browniebroke** [comment]
Ok, nice! I'm going  to close this now, I think you've reused the work through your other PR.

**Pull request closed without merging**